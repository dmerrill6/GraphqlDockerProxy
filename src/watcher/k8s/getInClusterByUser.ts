import { k8sUser, k8sUserPassword } from '../../properties';

export const getInClusterByUser = () => {

  const host = process.env.KUBERNETES_SERVICE_HOST;
  const port = process.env.KUBERNETES_SERVICE_PORT;

  return {
    url: 'https://' + host + ':' + port,
    auth: {
      user: k8sUser(),
      pass: k8sUserPassword(),
    },
    insecureSkipTlsVerify: true,
  };

};
