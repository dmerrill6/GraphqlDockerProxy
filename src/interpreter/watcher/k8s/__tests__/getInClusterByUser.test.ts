import { getInClusterByUser } from '../getInClusterByUser'
jest.mock('../../../../properties', () => {
  return {
    k8sUser: () => {
      return 'mockUser'
    },
    k8sUserPassword:() => {
      return 'mockPassword'
    },
  }

})
describe('tests the getInClusterByUser', () => {
  it('snapshot result', () => {
    process.env.KUBERNETES_SERVICE_HOST = 'mockHost'
    process.env.KUBERNETES_SERVICE_PORT = 'mockPort'
    expect(getInClusterByUser()).toMatchSnapshot()
  })
})
