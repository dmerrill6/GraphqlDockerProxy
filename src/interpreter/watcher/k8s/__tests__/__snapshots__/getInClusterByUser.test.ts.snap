// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the getInClusterByUser snapshot result 1`] = `
Object {
  "auth": Object {
    "pass": "mockPassword",
    "user": "mockUser",
  },
  "insecureSkipTlsVerify": true,
  "url": "https://mockHost:mockPort",
}
`;
